### PR TITLE
Fail builds if duplicate dependencies in pom

### PR DESCRIPTION
**Pull request by dbw9580**
### What changes are proposed in this pull request?

Fail builds if duplicate dependencies are detected in the pom.

### Why are the changes needed?

Make builds more reliable and easy to reason about.

### Does this PR introduce any user facing changes?

No.

### CODE REVIEW

**Comment by dbw9580** [comment]
To test this, duplicate any dependency from the `dependencies` section in a pom file. The build should fail at the `validate` phase of the build.

File: pom.xml [hunk 1]
```diff
@@ -1201,6 +1201,21 @@
...
+            <id>no-duplicate-dependencies</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <!-- checks for duplicate dependencies in the "dependencies" and "dependencyManagement" sections. -->
+                <!-- dependencies are considered duplicate if they have the same groupId, artifactId, type, and classifier properties -->
+                <!-- Note: enforcer v3.0.0-M3 does not correctly check "dependencyManagement" section for duplicates:
+                     see https://issues.apache.org/jira/browse/MENFORCER-301 -->
```
Comment (dbw9580) [inline pom.xml:1214]: Can we upgrade the enforcer plugin?

Comment (Xenorith) [reply pom.xml:1214]: i don't see any reason why we cannot upgrade it. the last time the maven enforcer plugin was upgraded was in https://github.com/Alluxio/alluxio/pull/11849

Comment (dbw9580) [reply pom.xml]: upgraded in #17912

**Review by Xenorith — APPROVED** [review]

**Review by yyongycy — APPROVED** [review]
lgtm

**Comment by dbw9580** [comment]
alluxio-bot, merge this please

**Comment by dbw9580** [comment]
alluxio-bot, merge this please

**Pull request merged**